### ISSUE TITLE
TestImageOperationsGolden fails on aarch64, ppc64le, s390x

### ISSUE THREAD

**Issue by QuLogic** [gohugoio/hugo#6446]
`TestImageOperationsGolden` was failing on these arches in 0.58.3 as reported in #6387, and should have been fixed by #6396. This is true: if I [build 0.58.3](https://koji.fedoraproject.org/koji/taskinfo?taskID=38473125), then it fails just like the original report, but [building 0.58.3 + a backported patch](https://koji.fedoraproject.org/koji/buildinfo?buildID=1403461) works. _However_, between that time and 0.59.0, it seems to have [broken again](https://koji.fedoraproject.org/koji/taskinfo?taskID=38472904). According to a bisect on a ppc64le test machine, this began with 4b286b9d2722909d0682e50eeecdfe16c1f47fd8:

```
github.com/gohugoio/hugo/resources
--- FAIL: TestImageOperationsGolden (6.52s)
    image_test.go:549: resize /a/gopher-hero8_huaa0cd7d2cfc14ff32a57f171896f2285_13327_200x0_resize_bge3e615_box_2.png
    image_test.go:549: resize /a/gopher-hero8_huaa0cd7d2cfc14ff32a57f171896f2285_13327_200x0_resize_q75_bge3e615_box_2.jpg
    image_test.go:549: resize /a/gradient-circle_huf3d35257a40a8d6f525263a856c5ecfd_20069_200x0_resize_bge3e615_box_2.png
    image_test.go:549: resize /a/gradient-circle_huf3d35257a40a8d6f525263a856c5ecfd_20069_200x0_resize_q75_bge3e615_box_2.jpg
    image_test.go:561: resize /a/sunset_hu59e56ffff1bc1d8d122b1403d34e039f_90587_200x100_resize_q75_box.jpg
    image_test.go:561: resize /a/sunset_hu59e56ffff1bc1d8d122b1403d34e039f_90587_600x0_resize_q75_box.jpg
    image_test.go:561: resize /a/sunset_hu59e56ffff1bc1d8d122b1403d34e039f_90587_200x0_resize_q50_r90_box.jpg
    image_test.go:569: fill /a/sunset_hu59e56ffff1bc1d8d122b1403d34e039f_90587_300x200_fill_q75_gaussian_smart1.jpg
    image_test.go:569: fill /a/sunset_hu59e56ffff1bc1d8d122b1403d34e039f_90587_100x100_fill_q75_box_center.jpg
    image_test.go:569: fill /a/sunset_hu59e56ffff1bc1d8d122b1403d34e039f_90587_300x100_fill_q75_nearestneighbor_topleft.jpg
    image_test.go:569: fill /a/sunset_hu59e56ffff1bc1d8d122b1403d34e039f_90587_400x200_fill_q75_box_bottomleft.jpg
    image_test.go:577: fit /a/sunset_hu59e56ffff1bc1d8d122b1403d34e039f_90587_300x200_fit_q75_linear.jpg
    image_test.go:606: filter: {{0 <nil>} 0xc0000100f8} /a/sunset_hu59e56ffff1bc1d8d122b1403d34e039f_90587_abf356affd7d70d6bec3b3498b572191.jpg
    image_test.go:606: filter: {{0 [6]} 0xc0002a33b8} /a/sunset_hu59e56ffff1bc1d8d122b1403d34e039f_90587_3efc2d0f29a8e12c5a690fc6c9288854.jpg
    image_test.go:606: filter: {{0 [50]} 0xc000010100} /a/sunset_hu59e56ffff1bc1d8d122b1403d34e039f_90587_6c5c12ac79d3455ccb1993d51eec3cdf.jpg
    image_test.go:606: filter: {{0 [100]} 0xc000010108} /a/sunset_hu59e56ffff1bc1d8d122b1403d34e039f_90587_352eb0101b7c88107520ba719432bbb2.jpg
    image_test.go:606: filter: {{0 [30]} 0xc000864510} /a/sunset_hu59e56ffff1bc1d8d122b1403d34e039f_90587_c36da6818db1ab630c3f87f65170003b.jpg
    image_test.go:606: filter: {{0 [10 -10 -10]} 0xc000010110} /a/sunset_hu59e56ffff1bc1d8d122b1403d34e039f_90587_0d1b300da7a815ed567b6dadb6f2ce5e.jpg
    image_test.go:606: filter: {{0 [240 50 100]} 0xc000010118} /a/sunset_hu59e56ffff1bc1d8d122b1403d34e039f_90587_9f00027c376fe8556cc9996c47f23f78.jpg
    image_test.go:606: filter: {{0 [1.5]} 0xc000864530} /a/sunset_hu59e56ffff1bc1d8d122b1403d34e039f_90587_17fd3c558d78ce249b5f0bcbe1ddbffb.jpg
    image_test.go:606: filter: {{0 [1 1 0]} 0xc0002a34e0} /a/sunset_hu59e56ffff1bc1d8d122b1403d34e039f_90587_3f1b1455c4a7d13c5aeb7510f9a6a581.jpg
    image_test.go:606: filter: {{0 [0.5 7]} 0xc000864540} /a/sunset_hu59e56ffff1bc1d8d122b1403d34e039f_90587_d30c10468b33df9010d185a8fe8f0491.jpg
    image_test.go:606: filter: {{0 [5]} 0xc0002a34f0} /a/sunset_hu59e56ffff1bc1d8d122b1403d34e039f_90587_30fc2aab35ca0861bf396d09aebc85a4.jpg
    image_test.go:606: filter: {{0 <nil>} 0xc000864560} /a/sunset_hu59e56ffff1bc1d8d122b1403d34e039f_90587_f6d8fe32ce3e83abf130e91e33456914.jpg
    image_test.go:606: filter: {{0 [22]} 0xc000010120} /a/sunset_hu59e56ffff1bc1d8d122b1403d34e039f_90587_7d9bc4700565266807dc476421066137.jpg
    image_test.go:606: filter: {{0 [32.5]} 0xc000864590} /a/sunset_hu59e56ffff1bc1d8d122b1403d34e039f_90587_de1fe6c0f40e7165355507d0f1748083.jpg
    image_test.go:613: filter all /a/sunset_hu59e56ffff1bc1d8d122b1403d34e039f_90587_cb45fcba865177290c89dc9f41d6ff7a.jpg
    image_test.go:561: resize /a/gohugoio8_hu7f72c00afdf7634587afaa5eff2a25b2_73538_200x100_resize_box_2.png
    image_test.go:561: resize /a/gohugoio8_hu7f72c00afdf7634587afaa5eff2a25b2_73538_600x0_resize_box_2.png
    image_test.go:561: resize /a/gohugoio8_hu7f72c00afdf7634587afaa5eff2a25b2_73538_200x0_resize_q50_r90_box_2.png
    image_test.go:569: fill /a/gohugoio8_hu7f72c00afdf7634587afaa5eff2a25b2_73538_300x200_fill_gaussian_smart1_2.png
    image_test.go:569: fill /a/gohugoio8_hu7f72c00afdf7634587afaa5eff2a25b2_73538_100x100_fill_box_center_2.png
    image_test.go:569: fill /a/gohugoio8_hu7f72c00afdf7634587afaa5eff2a25b2_73538_300x100_fill_nearestneighbor_topleft_2.png
    image_test.go:569: fill /a/gohugoio8_hu7f72c00afdf7634587afaa5eff2a25b2_73538_400x200_fill_box_bottomleft_2.png
    image_test.go:577: fit /a/gohugoio8_hu7f72c00afdf7634587afaa5eff2a25b2_73538_300x200_fit_linear_2.png
    image_test.go:606: filter: {{0 <nil>} 0xc0003d6030} /a/gohugoio8_hu7f72c00afdf7634587afaa5eff2a25b2_73538_9a86fee686dd5973923f5ef5c3b0bc74.png
    image_test.go:606: filter: {{0 [6]} 0xc000315318} /a/gohugoio8_hu7f72c00afdf7634587afaa5eff2a25b2_73538_8544b956dc08b714975ae52d4dcfdd78.png
    image_test.go:606: filter: {{0 [50]} 0xc0003d6038} /a/gohugoio8_hu7f72c00afdf7634587afaa5eff2a25b2_73538_9d4c2220235b3c2d9fa6506be571560f.png
    image_test.go:606: filter: {{0 [100]} 0xc0003d6040} /a/gohugoio8_hu7f72c00afdf7634587afaa5eff2a25b2_73538_7852bca7fb011b36d030e4d35d8e1d90.png
    image_test.go:606: filter: {{0 [30]} 0xc000864290} /a/gohugoio8_hu7f72c00afdf7634587afaa5eff2a25b2_73538_1bf2d9610b385893204d0a57ef8d1532.png
    image_test.go:606: filter: {{0 [10 -10 -10]} 0xc0003d6048} /a/gohugoio8_hu7f72c00afdf7634587afaa5eff2a25b2_73538_798ebb7a9e9dc7edd40e2832eb77e457.png
    image_test.go:606: filter: {{0 [240 50 100]} 0xc0003d6050} /a/gohugoio8_hu7f72c00afdf7634587afaa5eff2a25b2_73538_bac1f274c6786fdb63dd215df2226cd9.png
    image_test.go:606: filter: {{0 [1.5]} 0xc0008642e0} /a/gohugoio8_hu7f72c00afdf7634587afaa5eff2a25b2_73538_9660b4bf59aeb8ac8714d3e466af6197.png
    image_test.go:606: filter: {{0 [1 1 0]} 0xc0003153a0} /a/gohugoio8_hu7f72c00afdf7634587afaa5eff2a25b2_73538_4c320010919da2d8b63ed24818b4d8e1.png
    image_test.go:606: filter: {{0 [0.5 7]} 0xc0008642f0} /a/gohugoio8_hu7f72c00afdf7634587afaa5eff2a25b2_73538_84a8d324276a96584446750f06d04bd4.png
    image_test.go:606: filter: {{0 [5]} 0xc0003153b0} /a/gohugoio8_hu7f72c00afdf7634587afaa5eff2a25b2_73538_c1ced24877f4b1baf563997e33cadcfa.png
    image_test.go:606: filter: {{0 <nil>} 0xc000864310} /a/gohugoio8_hu7f72c00afdf7634587afaa5eff2a25b2_73538_41369feac467f9ecec9ef46911b04fa1.png
    image_test.go:606: filter: {{0 [22]} 0xc0003d6058} /a/gohugoio8_hu7f72c00afdf7634587afaa5eff2a25b2_73538_888208ddeeeb3dcfe84697903ddffe30.png
    image_test.go:606: filter: {{0 [32.5]} 0xc000864340} /a/gohugoio8_hu7f72c00afdf7634587afaa5eff2a25b2_73538_de67126dc370f606d57f2c229b3accab.png
    image_test.go:613: filter all /a/gohugoio8_hu7f72c00afdf7634587afaa5eff2a25b2_73538_c74bb417b961e09cf1aac2130b7b9b85.png
    image_test.go:561: resize /a/gohugoio24_huc57dd738f4724f4b341121e66fd85555_267952_200x100_resize_box_2.png
    image_test.go:561: resize /a/gohugoio24_huc57dd738f4724f4b341121e66fd85555_267952_600x0_resize_box_2.png
    image_test.go:561: resize /a/gohugoio24_huc57dd738f4724f4b341121e66fd85555_267952_200x0_resize_q50_r90_box_2.png
    image_test.go:569: fill /a/gohugoio24_huc57dd738f4724f4b341121e66fd85555_267952_300x200_fill_gaussian_smart1_2.png
    image_test.go:569: fill /a/gohugoio24_huc57dd738f4724f4b341121e66fd85555_267952_100x100_fill_box_center_2.png
    image_test.go:569: fill /a/gohugoio24_huc57dd738f4724f4b341121e66fd85555_267952_300x100_fill_nearestneighbor_topleft_2.png
    image_test.go:569: fill /a/gohugoio24_huc57dd738f4724f4b341121e66fd85555_267952_400x200_fill_box_bottomleft_2.png
    image_test.go:577: fit /a/gohugoio24_huc57dd738f4724f4b341121e66fd85555_267952_300x200_fit_linear_2.png
    image_test.go:606: filter: {{0 <nil>} 0xc0000100f8} /a/gohugoio24_huc57dd738f4724f4b341121e66fd85555_267952_f5d42d1797f90edd6379e0b082fdd53b.png
    image_test.go:606: filter: {{0 [6]} 0xc0002a3518} /a/gohugoio24_huc57dd738f4724f4b341121e66fd85555_267952_9c5c204a4fc82e861344066bc8d0c7db.png
    image_test.go:606: filter: {{0 [50]} 0xc000010100} /a/gohugoio24_huc57dd738f4724f4b341121e66fd85555_267952_d1ad299f68cb4b3e1eba2ab7633e7857.png
    image_test.go:606: filter: {{0 [100]} 0xc000010108} /a/gohugoio24_huc57dd738f4724f4b341121e66fd85555_267952_a0088abf33fdbf6be1651a71e7d4dc33.png
    image_test.go:606: filter: {{0 [30]} 0xc000864580} /a/gohugoio24_huc57dd738f4724f4b341121e66fd85555_267952_cdb3de8b01145d94ba41047655e42695.png
    image_test.go:606: filter: {{0 [10 -10 -10]} 0xc000010110} /a/gohugoio24_huc57dd738f4724f4b341121e66fd85555_267952_14fabac035a010e707ee3733f6590555.png
    image_test.go:606: filter: {{0 [240 50 100]} 0xc000010118} /a/gohugoio24_huc57dd738f4724f4b341121e66fd85555_267952_55b828db27003cb979bac711748f4789.png
    image_test.go:606: filter: {{0 [1.5]} 0xc0008645a0} /a/gohugoio24_huc57dd738f4724f4b341121e66fd85555_267952_d1f39c78ba8a0ada8233161edeed27ee.png
    image_test.go:606: filter: {{0 [1 1 0]} 0xc0002a3550} /a/gohugoio24_huc57dd738f4724f4b341121e66fd85555_267952_874d58b1c4b4b538f7ade152b3e57df8.png
    image_test.go:606: filter: {{0 [0.5 7]} 0xc0008645b0} /a/gohugoio24_huc57dd738f4724f4b341121e66fd85555_267952_621ae6f4010e2eb164521f54f653df1f.png
    image_test.go:606: filter: {{0 [5]} 0xc0002a3560} /a/gohugoio24_huc57dd738f4724f4b341121e66fd85555_267952_65ffdad1306cecec4d21bac1edd47c44.png
    image_test.go:606: filter: {{0 <nil>} 0xc0008645d0} /a/gohugoio24_huc57dd738f4724f4b341121e66fd85555_267952_958fee7992cf502355355c021148638b.png
    image_test.go:606: filter: {{0 [22]} 0xc000010120} /a/gohugoio24_huc57dd738f4724f4b341121e66fd85555_267952_dd36fa3cc8ae7cf4d686caf1a171284b.png
    image_test.go:606: filter: {{0 [32.5]} 0xc000864600} /a/gohugoio24_huc57dd738f4724f4b341121e66fd85555_267952_84b0614b9f84c94c0773ef49ae868d0b.png
    image_test.go:613: filter all /a/gohugoio24_huc57dd738f4724f4b341121e66fd85555_267952_cfc2eacca4b2748852f953954207d615.png
    image_test.go:657: resulting image differs from golden: gohugoio8_hu7f72c00afdf7634587afaa5eff2a25b2_73538_300x200_fill_gaussian_smart1_2.png
    image_test.go:655: expectedly differs from golden due to dithering: gohugoio8_hu7f72c00afdf7634587afaa5eff2a25b2_73538_4c320010919da2d8b63ed24818b4d8e1.png
    image_test.go:655: expectedly differs from golden due to dithering: gohugoio8_hu7f72c00afdf7634587afaa5eff2a25b2_73538_9d4c2220235b3c2d9fa6506be571560f.png
    image_test.go:655: expectedly differs from golden due to dithering: gohugoio8_hu7f72c00afdf7634587afaa5eff2a25b2_73538_c74bb417b961e09cf1aac2130b7b9b85.png
FAIL
exit status 1
FAIL	github.com/gohugoio/hugo/resources	7.989s
```

### What version of Hugo are you using (`hugo version`)?
0.59.0

### Does this issue reproduce with the latest release?
Yes.

**Comment by bep** [comment]
Yea, this was kind of expected. I think we need a test fix for this that is a little more robust.

**Comment by QuLogic** [comment]
Seems to be fixed in 0.59.1, probably by #6458.

**Comment by anthonyfok** [comment]
Hi @QuLogic, thank you for the report!

Sorry, I didn't see your report till just moments ago.  Hope that the use of Travis CI’s arm64 build (see PR #6467, committed to master as ae4fde0866b2a10f0a414e0d76c4ff09bed3776e) will help us catch similar cases sooner in the future.
